src/components/modals/export-data-modal.js

```diff
@@ -45,90 +46,122 @@ const propTypes = {
+
   return `${rowCount.toLocaleString()} Rows`;
 };
 
-const ExportDataModal = ({
-  datasets,
-  selectedDataset,
-  dataType,
-  filtered,
-  config,
-  // callbacks:
-  onChangeExportDataType,
-  onChangeExportSelectedDataset,
-  onChangeExportFiltered
-}) => (
-  <StyledModalContent className="export-data-modal">
-    <div>
-      <StyledExportSection>
-        <div className="description">
-          <div className="title">
-            Dataset
-          </div>
-          <div className="subtitle">
-            Choose the datasets you want to export
-          </div>
-        </div>
-        <div className="selection">
-          <select value={selectedDataset} onChange={e => onChangeExportSelectedDataset(e.target.value)}>
-          {['All'].concat(Object.keys(datasets)).map(d => (
-            <option key={d} value={d}>{(datasets[d] && datasets[d].label) || d}</option>
-          ))}
-          </select>
-        </div>
-      </StyledExportSection>
-      <StyledExportSection>
-        <div className="description">
-          <div className="title">
-            Data Type
-          </div>
-          <div className="subtitle">
-            Choose the type of data you want to export
-          </div>
-        </div>
-        <div className="selection">
-          {EXPORT_DATA_TYPE_OPTIONS.map(op =>
-            <StyledType
-              key={op.id}
-              selected={dataType === op.id}
-              available={op.available}
-              onClick={() => op.available && onChangeExportDataType(op.id)}
-            >
-              <FileType ext={op.label} height="80px" fontSize="11px" />
-            </StyledType>
-          )}
-        </div>
-      </StyledExportSection>
-      <StyledExportSection>
-        <div className="description">
-          <div className="title">
-            Filter Data
-          </div>
-          <div className="subtitle">
-            You can choose exporting original data or filtered data
-          </div>
-        </div>
-        <div className="selection">
-          <StyledFilteredOption selected={!filtered} onClick={() => onChangeExportFiltered(false)}>
-            <div className="filtered-title">Unfiltered Data</div>
-            <div className="filtered-subtitle">{getDataRowCount(datasets, selectedDataset, false)}</div>
-          </StyledFilteredOption>
-          <StyledFilteredOption selected={filtered} onClick={() => onChangeExportFiltered(true)}>
-            <div className="filtered-title">Filtered Data</div>
-            <div className="filtered-subtitle">{getDataRowCount(datasets, selectedDataset, true)}</div>
-          </StyledFilteredOption>
-        </div>
-      </StyledExportSection>
-    </div>
-  </StyledModalContent>
-);
+const ExportDataModalFactory = () => {
+  class ExportDataModal extends Component {
+    componentDidMount() {
+      const toCPUFilter =
+        this.props.selectedDataset || Object.keys(this.props.datasets);
+      this.props.applyCPUFilter(toCPUFilter);
```

Comment:
why are we applying CPUfilter when the component is mounted?

Reply:
Yes, we start cpu data filtering when someone click to open the export data modal. And in the cpu filter function, we compare `filterRecord` to decide if we need to run the actual filteirng. if not we use the previous filtered data